yaml_specs/modulemd_stream_v2.yaml

```diff
@@ -4,40 +4,66 @@ document: modulemd
 # Module metadata format version
 version: 2
 data:
-    # Module name, optional
-    # Typically filled in by the buildsystem, using the VCS repository
-    # name as the name of the module.
+    # Module name
+    # Filled in by the build system, using the VCS repository name as the name
+    # of the module.
+    #
+    # When included as input to the build system, the behavior is unspecified.
+    #
+    # Mandatory for module metadata in a yum/dnf repository.
     name: foo
-    # Module update stream, optional
-    # Typically filled in by the buildsystem, using the VCS branch name
-    # as the name of the stream.
+
+    # Module update stream
+    # Filled in by the buildsystem, using the VCS branch name as the name of
+    # the stream.
+    #
+    # When included as input to the build system, the behavior is unspecified.
+    #
+    # Mandatory for module metadata in a yum/dnf repository.
     stream: latest
-    # Module version, integer, optional, cannot be negative
-    # Typically filled in by the buildsystem, using the VCS commit
-    # timestamp.  Module version defines upgrade path for the particular
-    # update stream.
+
+    # Module version, integer, cannot be negative
+    # Filled in by the buildsystem, using the VCS commit timestamp.  Module
+    # version defines upgrade path for the particular update stream.
+    #
+    # When included as input to the build system, the behavior is unspecified.
+    #
+    # Mandatory for module metadata in a yum/dnf repository.
```

Comment:
I think this needs something noted about what the value `0` means. Since `version` is an integer, I think `0` is not allowed as a module version identifier, and gets treated treated the same as if it were unspecified.

Reply:
It's actually the other way around. Unspecified is the same as `version==0` which is a valid version. The special-case here is that if you input `0`, you won't see it in the output. I'll try to clear that up. I guess that means this field isn't *actually* mandatory, though.